internal/controller/provider_controller.go

```diff
@@ -245,9 +247,13 @@ func (r *ProviderReconciler) validateCredentials(ctx context.Context, provider *
 			return fmt.Errorf("failed to read secret, error: %w", err)
 		}
 
-		caFile, ok := secret.Data["caFile"]
+		caFile, ok := secret.Data["ca.crt"]
 		if !ok {
-			return fmt.Errorf("no caFile found in secret %s", provider.Spec.CertSecretRef.Name)
+			caFile, ok = secret.Data["caFile"]
+			if !ok {
+				return fmt.Errorf("no `ca.crt` key found in secret '%s'", provider.Spec.CertSecretRef.Name)
```

Comment:
```suggestion
				return fmt.Errorf("no 'ca.crt' nor 'caFile' key found in secret", "secret", provider.Spec.CertSecretRef.Name)
```

Reply:
I am not sure if we want to add the deprecated key `caFile` so as not to encourage users to use it.